### ISSUE TITLE
SD card reader not working on macOS 11+

### ISSUE THREAD

**Issue by SpookHCK** [osy/HaC-Mini#613]
So a strange issue... Probably hardware / firmware and not possible to fix, more wondering if anyone else sees this...
I have updated to 11.5; now 11.5.1 and all of a sudden my SD Card reader won't read SanDisks; adapted via the sandisk microSD to SD adapter... it is odd... only SanDisk micro SD cards, I have tried on another machine with the same O2 reader and that doesn't read it either... My windows box reads it fine but that uses a non-O2 reader...

I'm wondering if anyone else is seeing this... again only Sandisk and only after 11.5

I would call it a hardware failure, but  my other UHS-1 microSDHC card works fine (SP Elite) and an old ultra 2 SD card also works fine.

So.. let me know if you hit this, or if your "SanDisk microSDHC UHS-1" cards work via the SanDisk Adapter...

Thanks

**Comment by SpookHCK** [comment]
So even stranger... I just got a SD to USB adapter, and it worked, when I connected the micro sd card directly, so I then tried it via the adapter, and that worked, and then as if by magic, the adapter worked in the O2 bay, so I guess if you hit this, just try connect via a different interface and that may fix you up!

**Comment by SpookHCK** [comment]
yeah, so not really... It worked for a couple of times then the SanDisk O2 combo stopped again
The other brand SD card works in the O2 adapter just fine.. and the SanDisk cards wok just fine in the USB reader... Just not SanDisk and the internal O2... any suggestions or similar issues?

**Comment by KrisCris** [comment]
yes, the sd card slot doesn't work on my machine too. I don't use it very often so it's not a problem for me...but it would be nice to have it fixed

**Comment by osy** [comment]
To clarify both of you only see it broken in 11.5 and works on 11.4 and below?

**Comment by KrisCris** [comment]
> To clarify both of you only see it broken in 11.5 and works on 11.4 and below?

I've upgraded the OS to the latest version (11.5.2), but the SD card reader was already broken before I did that. I can't really recall what was the last version I used, but it is very likely to be 11.4 and below.

**Comment by osy** [comment]
Works fine for me on 11.5.1. I used a regular SD card. Does the SD slot work on Windows for you guys?

**Comment by osy** [comment]
Seems like other people are experiencing this as well. I wonder if it's an Apple driver issue. Can someone try injecting/replacing 10.15.7 AppleSDXC.kext and seeing if that work? I can't reproduce this issue on my machine but my SD cards are probably really old so they work fine.

**Comment by Simanias** [comment]
@osy I would love to test the 10.15.7 version, but I'm unable to get it. Any chance you have that version of the AppleSDXC.kext in your hands?

**Comment by osy** [comment]
@SpookHCK @Simanias Please test #965 (https://github.com/osy/HaC-Mini/suites/4447714551/artifacts/117989955) and report if it works. I've bundled the 11.4 SDXC drivers.

**Comment by Simanias** [comment]
@osy I can confirm it's working now, thanks for the quick fix!

**Comment by jitMatrix** [comment]
@osy Hi, my sd card reader still doesn't work well with 3.1.4(macOS 12.0.1). How to resolve the problem?

**Comment by jitMatrix** [comment]
> @osy Hi, my sd card reader still doesn't work well with 3.1.4(macOS 12.0.1). How to resolve the problem?

oh, I double checked on it. It seems 32GB(HC-1) can work well but 128GB(XC-1) cannot. They were both work well on Catalina.

**Comment by osy** [comment]
Can you double check with a new install of Catalina? I don't think XC ever worked for me.
